client/rest-high-level/src/main/java/org/elasticsearch/client/RestHighLevelClient.java

```diff
@@ -447,6 +448,20 @@ public final BulkByScrollResponse reindex(ReindexRequest reindexRequest, Request
         );
     }
 
+    /**
+     * Submits a reindex task.
+     * See <a href="https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-reindex.html">Reindex API on elastic.co</a>
+     * @param reindexRequest the request
+     * @param options the request options (e.g. headers), use {@link RequestOptions#DEFAULT} if nothing needs to be customized
+     * @return the submission response
+     * @throws IOException in case there is a problem sending the request or parsing back the response
+     */
+    public final TaskId submitReindexTask(ReindexRequest reindexRequest, RequestOptions options) throws IOException {
```

Comment:
The name submit is breaking the API spec contract. We have test that verifies if we are not adding any additional methods apart from those from the API spec. 

`
RestHighLevelClientTests.testApiNamingConventions <<< FAILURES!
19:16:19    > Throwable #1: java.lang.AssertionError: Some client method doesn't match a corresponding API defined in the REST spec: [submit_reindex_task]
19:16:19    > Expected: <0>
19:16:19    >      but: was <1>
`
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+pull-request/1777/console

Reply:
Yeah I just hit that one too trying to return a GetTaskResponse wrapped in an Optional (following [discussion](https://github.com/elastic/elasticsearch/issues/27205#issuecomment-433521218) on how Java clents should handle 404s)

Reply:
I think maybe you can teach it about these sorts of names. They aren't required for every method, but maybe we could make them required if `wait_for_completion` was a parameter in the spec.